x-pack/solutions/observability/plugins/observability/public/components/alert_actions/inverted_alert_actions.tsx

```diff
@@ -0,0 +1,280 @@
+    }, []);
+
+    const handleViewInAppUrl = useCallback(() => {
+      const alertLink = observabilityAlert.link as unknown as string;
+      if (!observabilityAlert.hasBasePath) {
+        setViewInAppUrl(prepend(alertLink ?? ''));
+      } else {
+        setViewInAppUrl(alertLink);
+      }
+    }, [observabilityAlert.link, observabilityAlert.hasBasePath, prepend]);
+
+    const { isPopoverOpen, setIsPopoverOpen, handleAddToExistingCaseClick, handleAddToNewCaseClick } =
+      useCaseActions({
+        refresh,
+        alerts: [alert],
+        services: {
+          cases,
+        },
+      });
+
+    const closeActionsPopover = useCallback(() => {
+      setIsPopoverOpen(false);
+    }, [setIsPopoverOpen]);
+
+    const toggleActionsPopover = () => {
+      setIsPopoverOpen(!isPopoverOpen);
+    };
+
+    const actionsMenuItems = [
+      ...(userCasesPermissions?.createComment && userCasesPermissions?.read
+        ? [
+            <EuiContextMenuItem
+              data-test-subj="add-to-existing-case-action"
+              key="addToExistingCase"
+              onClick={handleAddToExistingCaseClick}
+              size="s"
+            >
+              {i18n.translate('xpack.observability.alerts.actions.addToCase', {
+                defaultMessage: 'Add to existing case',
+              })}
+            </EuiContextMenuItem>,
+            <EuiContextMenuItem
+              data-test-subj="add-to-new-case-action"
+              key="addToNewCase"
+              onClick={handleAddToNewCaseClick}
+              size="s"
+            >
+              {i18n.translate('xpack.observability.alerts.actions.addToNewCase', {
+                defaultMessage: 'Add to new case',
+              })}
+            </EuiContextMenuItem>,
+          ]
+        : []),
+      useMemo(
+        () => (
+          <DefaultAlertActions<ObservabilityAlertsTableContext>
+            observabilityRuleTypeRegistry={observabilityRuleTypeRegistry}
+            key="defaultRowActions"
+            onActionExecuted={closeActionsPopover}
+            isAlertDetailsEnabled={true}
+            resolveRulePagePath={(ruleId, currentPageId) =>
+              currentPageId !== RULE_DETAILS_PAGE_ID ? paths.observability.ruleDetails(ruleId) : null
+            }
+            resolveAlertPagePath={(alertId, currentPageId) =>
+              currentPageId !== ALERT_DETAILS_PAGE_ID
+                ? paths.observability.alertDetails(alertId)
+                : null
+            }
+            tableId={tableId}
+            refresh={refresh}
+            alert={alert}
+            openAlertInFlyout={openAlertInFlyout}
+            services={services}
+            {...rest}
+          />
+        ),
+        [
+          alert,
+          closeActionsPopover,
+          observabilityRuleTypeRegistry,
+          openAlertInFlyout,
+          refresh,
+          services,
+          rest,
+          tableId,
+        ]
+      ),
+      ...extraRowActions.map((action) => (
```

Comment:
Here's where I append the extra actions -- I'm repeating all of the `<EuiContextMenuitem>` stuff here when I'd rather there just be a component that can render those items the way they're supposed to be rendered, but this is fine for what I'm trying to convey in this quick demo.